lib/signed-xml.js

```diff
@@ -377,8 +383,8 @@ SignedXml.prototype.checkSignature = function (xml, callback) {
     }
   }
 
-  this.signingKey = this.keyInfoProvider.getKey(this.keyInfo);
-  if (!this.signingKey) {
+  this.signingCert = this.keyInfoProvider.getCert(this.keyInfo);
```

Comment:
This is confusing.
It should be `this.signingCert = this.keyInfoProvider.getCert(this.keyInfo);`
And then in `validateSignatureValue` also change to `this.signingCert`:

Overall more clear lines of separation are needed in the library between what the private signing key is and what the public certificate is.

Reply:
I don't follow your concern. It seems that it is already as you suggest. On line 478, we pass `this.signingCert || this.signingKey` because the downstream functions will accept a public key or a private key, because a public key can be derived from a private key. Since we aren't working with "public keys" directly in this codebase, we're sending `signingCert` instead, since a cert only contains the public part.

Reply:
We already have these two properties in SignedXml. We have `this.signingKey`, which is meant to contain the private key, and we have `this.signingCert`, which contains the certificate (public key).
So why put the public key you get from `keyInfoProvider.getCert(..)` into the property which is meant to contain the private key?

Reply:
I don't. The line as it current is reads: `this.signingCert = this.keyInfoProvider.getCert(this.keyInfo);` The previous name was very confusing on this matter, thus my correcting it. Can you clarify further what you're talking about? Maybe there is a line that I'm misreading.